### ISSUE TITLE
Add Support for Notes

### ISSUE THREAD

**Issue by mattangus** [iterative/studio-support#29]
I'm not sure how this would be implemented or if it's possible. But It would be greate to have git notes show up through the UI. Ideally the would be editable as well. Documentation of `git notes` can be found [here](https://git-scm.com/docs/git-notes). This would make tracking experiments through studio a more holistic experience. git notes have some advantages over tags (limited in length and formatting) an commit messages (immutable).

**Comment by shcheklein** [comment]
Hey, @mattangus , thanks for the request. Do you already use them in your workflow? Or is it an idea how to support "distributed per-commit comments" that team can do and share?

**Comment by mattangus** [comment]
Yes it's more about distributed per commit comments, but I was trying to emphasize the adding/editing functionality. Since for example later experiments can teach you things about previous ones and the notes could be updated to reflect that!

**Comment by shcheklein** [comment]
Yep, I thinks I understand the use case. I'm trying to better understand if we should use git notes for this or some SaaS storage that we already have in Studio. Do you have any specific pros in mind (or if you already use them) on using git notes for this?

**Comment by mattangus** [comment]
I don't currently use them. The only benefit I see is that the sources of information would be consolidated, and accessible from the command line.
